src/lib/api/basyx-v3/api.ts

```diff
@@ -540,6 +653,41 @@ export const SubmodelRepositoryApiFp = function (configuration?: Configuration)
                 }
             };
         },
+
+        /**
+         * @summary Uploads an attachment to a specific submodel element.
+         * @param {string} submodelId - The unique identifier of the submodel containing the submodel element.
+         * @param {AttachmentDetails} attachmentDetails - The attachment data to be uploaded to the submodel element.
+         * @param {object} [options] - Optional. Additional options to override default HTTP request settings.
+         * @throws {RequiredError}
+         */
+        putAttachmentToSubmodelElement(submodelId: string, attachmentDetails: AttachmentDetails, options: any) {
+            return async (requestHandler: FetchAPI = isomorphicFetch, basePath: string = BASE_PATH) => {
```

Comment:
I believe this method is autogenerated, but I think it's kind of confusing that the method doesn't actually put the thubmanail to a show, but returns a callback function that does so. I think the method was easier to use and understand if you just directly perform the put inside the method. Or maybe there is a good reason for why it's like this that I don't understand :/.